### PR TITLE
Split validation without packing full data into a list

### PR DESCRIPTION
**Pull request by SunQpark**
I have discovered that data loader of this template packs all data into a python list, to split validation set. Although doing this is not a problem when dealing with small datasets like mnist, it is not suitable for normal datasets which are too big to be contained into memory. I replaced this with torch.utils.data.sampler.SubsetRandomSampler which specifies indices to be selected in a data loader.

By the way, I have been using this template for about 3 months changing things, so this request will contain  some unimportant small changes. Sorry for not organizing things before pulling request.

### CODE REVIEW

**Comment by victoresque** [comment]
Hi,

Nice work, thanks for your contribution! Since the project has changed to using `.json` configs (instead of `argparse`), I'll try to merge your changes into the current project, keep up the great work!

Best,
Victor

**Comment by SunQpark** [comment]
I have made another PR #10 that has no conflict. please merge that and I will close this PR
